utils/utils_test.go

```diff
@@ -1,10 +1,195 @@
 package utils
 
 import (
+	"context"
+	"github.com/go-sql-driver/mysql"
+	"github.com/lib/pq"
 	"github.com/stretchr/testify/require"
 	"testing"
+	"time"
 )
 
+func TestBatchSliceOfStrings(t *testing.T) {
+	subtests := []struct {
+		name   string
+		keys   []string
+		count  int
+		output [][]string
+	}{
+		{"nil", nil, 1, nil},
+		{"empty", make([]string, 0, 1), 1, nil},
+		{"a", []string{"a"}, 1, [][]string{{"a"}}},
+		{"a2", []string{"a"}, 2, [][]string{{"a"}}},
+		{"a_b", []string{"a", "b"}, 1, [][]string{{"a"}, {"b"}}},
+		{"ab", []string{"a", "b"}, 2, [][]string{{"a", "b"}}},
+		{"ab3", []string{"a", "b"}, 3, [][]string{{"a", "b"}}},
+		{"a_b_c", []string{"a", "b", "c"}, 1, [][]string{{"a"}, {"b"}, {"c"}}},
+		{"ab_c", []string{"a", "b", "c"}, 2, [][]string{{"a", "b"}, {"c"}}},
+		{"abc", []string{"a", "b", "c"}, 3, [][]string{{"a", "b", "c"}}},
+		{"abc4", []string{"a", "b", "c"}, 4, [][]string{{"a", "b", "c"}}},
+	}
```

Comment:
As the `count` parameter is a signed integer, try negative numbers :smirk: 

I would suggest making the function documentation more obvious and stating that `count <= 0` will either panic or be rewritten to 1. A `count` of 0 returns an infinite loop emitting empty string arrays until eternity and a negative `count` results in a slice bounds out of range access.

```go
{"abc0-to-infinity-and-beyond", []string{"a", "b", "c"}, 0, [][]string{{}, {}, {}}},
{"abc-1-crash", []string{"a", "b", "c"}, -1, nil},
```

```
=== RUN   TestBatchSliceOfStrings/abc0-to-infinity-and-beyond
    utils_test.go:51: 
        	Error Trace:	/home/REDACTED/git/github.com/Icinga/icinga-go-library/utils/utils_test.go:51
        	Error:      	Should be false
        	Test:       	TestBatchSliceOfStrings/abc0-to-infinity-and-beyond
        	Messages:   	receiving from channel should not return anything, received []
=== RUN   TestBatchSliceOfStrings/abc-1-crash
panic: runtime error: slice bounds out of range [:-1]
```

Reply:
Where I just ref an issue/PR (like you https://github.com/Icinga/icinga-go-library/pull/32#discussion_r1718314038), I prefix it with * and a space, so I show its title and status. E.g.:

* #66

Reply:
After this got merged, please include it in the tests, e.g., by adding another `shouldPanic` field to the table test struct.